src/Broadway/Domain/Metadata.php

```diff
@@ -43,24 +56,37 @@ public function merge(Metadata $otherMetadata)
     }
 
     /**
-     * {@inheritDoc}
+     * Returns an array with all metadata.
+     *
+     * @return array
      */
-    public function serialize()
+    public function all()
     {
         return $this->values;
     }
 
     /**
-     * Helper method to construct an instance containing the key and value.
+     * Get a specific metadata value based on key.
      *
-     * @param mixed $key
-     * @param mixed $value
+     * @param string $key
      *
-     * @return Metadata
+     * @return mixed
      */
-    public static function kv($key, $value)
+    public function get($key)
```

Comment:
Using `?:` or `??` (if using PHP7) encourages lazy evaluation, whereas passing default value to the function requires evaluating it beforehand.

It also prevents nesting:
```php
$metadata->get('foo', $metadata->get('bar', $metadata->get('baz')));

$metadata->get('foo') ?? $metadata->get('bar') ?? $metadata->get('baz');
```